docs/changelog/1624.bugfix.rst

```diff
@@ -0,0 +1 @@
+Discover python interpreter in a case insensitive manner
```

Comment:
please add your user name as: `` - by :user:`PrajwalM2212` ``

You'll also need to add test for this to confirm this behaviour; probably by extending https://github.com/pypa/virtualenv/blob/master/tests/unit/discovery/py_info/test_py_info.py#L66